lib/widgets/post/post_video.dart

```diff
@@ -0,0 +1,110 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:video_player/video_player.dart';
+import '../../util/redgif.dart' as redgif;
+
+//TODO Support for full screen video
+
+class PostVideo extends StatefulWidget {
+  final Uri url;
+  const PostVideo(this.url, {super.key});
+
+  @override
+  State<PostVideo> createState() => _PostVideoState(url);
+}
+
+class _PostVideoState extends State<PostVideo> {
+  late VideoPlayerController _controller;
+  late Future<void> _initializeVideoPlayerFuture;
+  final Uri url;
+  _PostVideoState(this.url);
+
+  @override
+  void initState() {
```

Comment:
So, instead of having all this code, you can just use the fancy new hooks:
https://github.com/rrousselGit/flutter_hooks#existing-hooks